web/tests/functional/report_viewer_api/__init__.py

```diff
@@ -91,7 +93,9 @@ def setup_class_common(workspace_name):
     # checkers to have some real difference.
     codechecker_cfg['checkers'] = ['-e', 'core.CallAndMessage',
                                    '-d', 'core.StackAddressEscape',
-                                   '-d', 'unix.Malloc'
+                                   '-d', 'unix.Malloc',
+                                   '-d', 'clang-diagnostic',
+                                   '-e', 'clang-diagnostic-division-by-zero'
```

Comment:
Why is this needed? I am more interested in the _design_ reason, not the technical one. Suppose that you are new to this test file, and stumble upon this invocation -- what does it convey to you? We manually disable all warnings for some reason, but enable a specific one after that?

Was this an attempt to reduce the amount of changes needed in the tests, otherwise everything broke?

Reply:
Same answer as above. As for the design reason: in this test we have multiple analyses with different configurations. Here we want to test which checkers' reports have appeared and disappeared. We control this by explicitly enabling and disabling them in different analysis actions.